src/FsAutoComplete.Core/SignatureFormatter.fs

```diff
@@ -22,18 +22,25 @@ module SignatureFormatter =
         | true, false -> b
         | false, false -> a + " " + b
 
-    let getUnioncaseSignature displayContext (unionCase:FSharpUnionCase) =
-        if unionCase.UnionCaseFields.Count > 0 then
-            let typeList =
-                unionCase.UnionCaseFields
-                |> Seq.map (fun unionField ->
-                    if unionField.Name.StartsWith "Item" then //TODO: Some better way of dettecting default names for the union cases' fields
-                        unionField.FieldType.Format displayContext
-                    else
-                        unionField.Name ++ ":" ++ ((unionField.FieldType.Format displayContext)))
+    let rec formatFSharpType (context: FSharpDisplayContext) (typ: FSharpType) : string =
+        try
+            if typ.IsTupleType || typ.IsStructTupleType then
+                typ.GenericArguments
+                |> Seq.map (formatFSharpType context)
                 |> String.concat " * "
-            unionCase.DisplayName + " of " + typeList
-         else unionCase.DisplayName
+            elif typ.GenericArguments.Count > 0 then
+                let genericArgs =
+                    typ.GenericArguments
+                    |> Seq.map (formatFSharpType context)
+                    |> String.concat ","
+                sprintf "%s<%s>" (PrettyNaming.QuoteIdentifierIfNeeded typ.TypeDefinition.DisplayName) genericArgs
+
+            elif typ.IsGenericParameter then
+                (if typ.GenericParameter.IsSolveAtCompileTime then "^" else "'") + typ.GenericParameter.Name
+            else
+                PrettyNaming.QuoteIdentifierIfNeeded (typ.Format context)
+        with
+        | _ -> typ.Format context
```

Comment:
This is a helper similar to the one in the docformatter its similar but not entirely the same ;-)